src/upload-button.tsx

```diff
@@ -231,10 +297,14 @@ export const UploadButton = ({
                             [file.name]: { ...oldFile, progress },
                         };
                     });
-                }, abort.signal);
-                // TODO: pass { superuser: try } depending on directory owner
+                }, abort.signal, options);
             } catch (exc) {
                 cockpit.assert(exc instanceof Error, "Unknown exception type");
+
+                // Clean up touched file
+                if (owner) {
+                    await cockpit.spawn(["rm", "-f", destination], { superuser: "require" });
```

Comment:
This can fail. Also, isn't it a bit more elegant to call `cockpit.file.replace(null)` here? (Same above)

Reply:
Does that work without a tag?

Reply:
Sure, .replace() never needs a tag. You can specify one for protecting against changes in the background, but if that is a temp file we don't need to care.